packages/core-sdk/lib/Connectors/Wallets/WalletConnectConnector.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  InvalidNetworkError,
+  RejectedByUser,
+  WalletNotDetectedError,
+} from "../../Errors";
+import { ethers, utils } from "ethers";
+import { BaseConnector } from "../BaseConnector";
+import { hexToDec } from "@unifiprotocol/utils";
+import { Web3BaseAdapter } from "../../Adapters/Web3BaseAdapter";
+import WalletConnectProvider from "@walletconnect/ethereum-provider";
+
+import { Web3MulticallAdapter } from "../../Adapters";
+import {
+  IConnectorAdapters,
+  IBlockchainConfig,
+  IAdapter,
+  IConnectorMetadata,
+  WalletConnectors,
+} from "../../Types";
+import { ForceNetworkError } from "../../Errors/ForceNetworkError";
+
+export class WalletConnectConnector extends BaseConnector {
+  private agent: WalletConnectProvider;
+  constructor(
+    config: IBlockchainConfig,
+    metadata: IConnectorMetadata = {
+      name: WalletConnectors.WalletConnect,
+      displayName: "WalletConnect",
+      isWallet: true,
+    }
+  ) {
+    super(metadata, config);
+    this.agent = this._initAgent();
```

Comment:
I guess the wallet/agent is always available because we inject it from their `lib`, right?

Reply:
Yep, exactly